MandelaK/microblog
# Chapter6

This covers chapter 6 of the tutorial. Here we implement the profile section of our website. Users can visit their profile and edit information about themselves. Their last seen is also displayed on their profile page. Users can only edit their profiles when they are logged in. Below are some of the things I've learnt:
- how to automatically send 404 error when a resource is not found
- how to import and display icons from gravatar
- how to use sub-templates instead of repeating code in separate template files
- a basic implementation 'last seen' functionality, whereby the last seen is captured every time a user sends a request to the server
- got a better understanding of migrations and their importance after making changes to our database tables
- how to render TextArea element using jinja2


### How can this be manually tested?
Follow installation instructions then check out to chapter6. Run the application and sign up new users or sign in any existing users. You can visit their profile by going to `../users/<username>`. You will get error page if you try to access the profile page of users who do not exist. If you are not logged in, you cannot view any profile. If you are signed in, you can edit your own profile if you visit the profile page, but you cannot edit profile of other users
